internal/pkg/runtime/engines/singularity/cleanup_linux.go

```diff
@@ -58,5 +63,47 @@ func (engine *EngineOperations) CleanupContainer(fatal error, status syscall.Wai
 		return file.Delete()
 	}
 
+	if engine.EngineConfig.CryptDev != "" {
+		cleanupCrypt(engine.EngineConfig.CryptDev)
+	}
+
+	return nil
+}
+
+func cleanupCrypt(path string) error {
+
+	// Elevate the privilege to unmount and delete the crypt device
+	runtime.LockOSThread()
+	uid := os.Getuid()
+	err := syscall.Setresuid(uid, 0, uid)
+	defer syscall.Setresuid(uid, uid, 0)
+	defer runtime.UnlockOSThread()
+	if err != nil {
+		sylog.Debugf("Err setting suid")
+		return err
```

Comment:
I think it would also be good to add more debugging in the error message:

```suggestion
		return fmt.Errorf("unable to setup suid: %v", err)
```

_Or something like that..._